### PR TITLE
Configurable date range

### PR DESCRIPTION
**Pull request by tsujigiri**

### CODE REVIEW

File: app/cells/date_range_cell.rb [hunk 1]
```diff
@@ -20,16 +20,22 @@ def date_field(type)
...
+      .merge(start: "##{id}_from", end: "##{id}_to")
+      .merge(options.slice(:start_date, :end_date).transform_values(&:to_s))
+  end
+
   def id
     "date_range_#{name_param}"
   end
+
+  def serialize(value)
+    ERB::Util.html_escape(value.to_json)
```
Comment (dziemba) [inline app/cells/date_range_cell.rb]: Is this really necessary?

```ruby
> content_tag(:div, '', test: '"')
=> "<div test=\"&quot;\"></div>"
> content_tag(:div, '', test: '\'')
=> "<div test=\"&#39;\"></div>"
```

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: The `html_escape` seems to be skipped. No clue why.

File: spec/components/date_range_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    it 'renders a label' do
+      expect(subject.css('label')).to be_present
+      expect(subject.css('label').text).to eq('Foo')
+    end
+  end
+
+  context 'without label' do
+    let(:date_range) { helper.ui_component('date_range', form: form, name: 'bar') }
+
+    it 'renders a label' do
```
Comment (dziemba) [inline spec/components/date_range_spec.rb]: ?

File: app/cells/date_range_cell.rb (hunk 1 above)
Comment (dziemba) [inline app/cells/date_range_cell.rb]: Skipped where? Can you find out?

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: Awesome! cells-slim overwrites `content_tag_string` and sets `escape` to `false`:
https://github.com/trailblazer/cells-slim/blob/1cba347250273ddd00177ed4ff64ad5cf89680a6/lib/cell/slim.rb#L61-L63

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: The best part is, that it keeps the default value of `true` for `escape` before doing so...

File: spec/components/date_range_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+RSpec.describe 'date_range', type: :helper do
+  subject do
+    Nokogiri::HTML.parse(date_range)
+  end
+
+  let(:form) do
+    helper.bootstrap_form_for :foo, url: '/' do |f|
+      return f
```
Comment (dziemba) [inline spec/components/date_range_spec.rb]: Can you explain this a bit?

File: app/cells/date_range_cell.rb (hunk 1 above)
Comment (dziemba) [inline app/cells/date_range_cell.rb]: Oh wow, that's not good. But nice idea with `escape_once`.

File: spec/components/date_range_spec.rb (hunk 3 above)
Comment (tsujigiri) [inline spec/components/date_range_spec.rb]: The intention was to escape the form's block, returning the form object to the calling scope. The thing is, that the calling scope is itself a block, so the return should break out of this too, should it? So, now that you ask, I really cannot explain why this actually works...

File: app/cells/date_range_cell.rb (hunk 1 above)
Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: Now it works on my machine, but not on CircleCI...

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: ...where it is now escaped twice:
```html
data-ranges="{&amp;quot;Next 10 days&amp;quot;:[&amp;quot;2015-06-22&amp;quot;,&amp;quot;2015-07-01&amp;quot;]}"
```

Comment (dziemba) [inline app/cells/date_range_cell.rb]: What about `CGI.escapeHTML`?

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: Unless the second time it is actually escaped by CircleCI... :D

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: `cells-slim` was a different version on CircleCI and on my machine. I just did `bundle update cells-slim` and it updated from 0.0.2 to 0.0.3, which it already is on CirlceCI. How is this even possible?

Comment (dziemba) [inline app/cells/date_range_cell.rb]: bundler does not care about the lockfile unless you use the `--deployment` flag

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: I think it does. The issue here is due to the fact that the Gemfile.lock isn't in the repo in the first place.

Comment (dziemba) [inline app/cells/date_range_cell.rb]: Okay, apparently they changed the behavior, it was weird anyway.

Not checking in the `.lock` file is correct, the proper solution would be to specify the `>0.0.3` dependency in the `.gemspec`. But keeping all that up to date is tedious, so let's just "rebuild w/o cache" on CircleCI when this happens again ;)

Comment (tsujigiri) [inline app/cells/date_range_cell.rb]: When we know this particular gem doesn't work as expected below that particular version, specifying the minimum version is a pretty decent idea, I think.

File: spec/components/date_range_spec.rb (hunk 3 above)
Comment (tsujigiri) [inline spec/components/date_range_spec.rb]: I guess it's returning from wherever this is evaluated and nothing crucial happens after that point in that method... So, it's kind of assuming this, which is not that nice. On the other hand it's a test, so it is tested behaviour. ;)

Comment (dziemba) [inline spec/components/date_range_spec.rb]: We should really rethink open-sourcing this... :grinning:

Comment (tsujigiri) [inline spec/components/date_range_spec.rb]: lol, yes.

Comment (tsujigiri) [inline spec/components/date_range_spec.rb]: One could also argue, though, that passing the form object to a local variable first and returning this from the let block, assumes that `bootstrap_form_for` doesn't change the object after the block. The interesting question now is, would this also happen with the `return`?

**Comment by tsujigiri** [comment]
Anything else?

**Pull request merged**